packages/@react-aria/selection/src/useSelectableItem.ts

```diff
@@ -281,6 +295,13 @@ export function useSelectableItem(options: SelectableItemOptions): SelectableIte
       ) {
         onSelect(e);
       }
+
+      // TODO: non ideal since this makes the link action happen on press start rather than on press up like it usually
+      // does for listboxes but we never get the onPress event because the simiulated event target from useAutocomplete is different
+      // from the actual place it is triggered from (the input).
+      if (shouldUseVirtualFocus && e.pointerType === 'keyboard' && hasAction) {
+        performAction(e);
+      }
```

Comment:
Not sure if there is a better way to go about this here. As described above the simulated event from useAutocomplete doesn't actually cause `onPress` to be triggered because the tracked `state.target` (listbox item) in usePress doesn't match/contain the event target's original target (the autocomplete input) but ideally we'd want links to be triggered on key up like they usually are for listboxes

Reply:
Hm, would there be issues with moving the focus briefly with `requestAnimationFrame()`, similar to what we do with `tabindex` in `useGridCell`?

Reply:
As in moving focus to the virtually focused item on `Enter` press, then dispatching the keydown event so that the event target for both the keydown/up is the focused item? That can certainly work but it does mean there will be a blur event on the input element that could be unexpected for the end user since this is supposed to be virtual focus. 

Additionally, if there any cases where we'd want to return focus back to the input after the focus movement to the item, it might lead to some unexpected/noisy screenreader announcements in addition to these extra blur events, will have to test. As an aside, supporting wrapping a GridList/Table is also going to be interesting because we'd want grid navigation to work but left/right arrow is going to also move the text cursor in the field and clear virtual focus for NVDA announcements... It may turn out that ditching virtual focus is easier tbh, and perhaps the user should just tab between the input and the wrapped collection

Reply:
Yep, I intended to always reset the focus back to the input on the next frame :thumbsup:

The blur events would indeed be an annoying side-effect, but we might be able to early return if we delay `onBlur` callback invocations to the next frame as well. Surely feels like fighting an uphill battle though.

Very interesting that autocomplete is planned to make its way over to grids eventually as well. Will follow the development on this closely :smile: